modules/Synapse/workspaces/integrationRuntimes/main.bicep

```diff
@@ -0,0 +1,63 @@
+@description('Conditional. The name of the parent Synapse Workspace. Required if the template is used in a standalone deployment.')
+param workspaceName string
+
+@description('Required. The name of the Integration Runtime.')
+param name string
+
+@allowed([
+  'Managed'
+  'SelfHosted'
+])
+@description('Required. The type of Integration Runtime.')
+param type string
+
+@description('Optional. The name of the Managed Virtual Network if using type "Managed" .')
+param managedVirtualNetworkName string = ''
+
+@description('Optional. Integration Runtime type properties. Required if type is "Managed".')
+param typeProperties object = {}
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = true
+
+resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (enableDefaultTelemetry) {
+  name: 'pid-47ed15a6-730a-4827-bcb4-0fd963ffbd82-${uniqueString(deployment().name)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
+
+var managedVirtualNetworkVar = {
+  referenceName: type == 'Managed' ? managedVirtualNetworkName : null
+  type: type == 'Managed' ? 'ManagedVirtualNetworkReference' : null
+}
+
+resource workspace 'Microsoft.Synapse/workspaces@2021-06-01' existing = {
+  name: workspaceName
+}
+
+resource integrationRuntime 'Microsoft.Synapse/workspaces/integrationRuntimes@2021-06-01' = {
+  name: name
+  parent: workspace
+  properties: type == 'Managed' ? {
+    type: type
+    managedVirtualNetwork: managedVirtualNetworkVar
```

Comment:
I'd suggest to move the above variable here.

Also if you do that it appears the dedicated `type == 'Manged'` checks inside the variables are redundant, right? The entire properties block will only be used if `type == Managed` anyways

Reply:
removed the conditions in the variable block

Reply:
Thanks @tyconsulting - I'd anyways suggest to also move the variable's content down in to the properties block as it is only used in one location is is very simple.

Aside from that I believe we can merge :) Thank you

Reply:
Just to make sure we're on the same page, I mean it like

```bicep
(...)
managedVirtualNetworkL {
  referenceName: 'default'
  type: 'ManagedVirtualNetworkReference'
}
(...)
```

Reply:
done. I initially copied this module from the ADF IR module because they are largely the same. you might want to update the ADF IR module as well.